### ISSUE TITLE
Runtime error in `layernorm + add + dropout` benchmarks

### ISSUE THREAD

**Issue by Priya2698** [NVIDIA/Fuser#1608]
The following error is seen for several input sizes in PR #1606. The error is seen when running the whole benchmark and may not be reproducible with individual sizes. 

Repro:
```
ERROR    nvfuser:__init__.py:196 An error occurred while executing nvFuser FusionDefinition 0.
If you believe this is a bug or need assistance, please file an issue at https://github.com/NVIDIA/Fuser/issues/new
Here's a script to reproduce the error:
```
```python
import torch
from nvfuser import FusionDefinition, DataType

def nvfuser_fusion_id0(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(shape=[-1, -1], contiguity=[True, True], dtype=DataType.Bool, is_cpu=False)
    T1 = fd.define_tensor(shape=[-1], contiguity=[True], dtype=DataType.Float, is_cpu=False)
    T2 = fd.define_tensor(shape=[-1, 1], contiguity=[True, None], dtype=DataType.Float, is_cpu=False)
    T3 = fd.define_tensor(shape=[-1, -1], contiguity=[True, True], dtype=DataType.Float, is_cpu=False)
    T4 = fd.define_tensor(shape=[-1], contiguity=[True], dtype=DataType.Float, is_cpu=False)
    T5 = fd.define_tensor(shape=[-1, -1], contiguity=[True, True], dtype=DataType.Float, is_cpu=False)
    T6 = fd.ops.mul(T5, T0)
    S7 = fd.define_scalar(1.11111, dtype=DataType.Double)
    T8 = fd.ops.mul(T6, S7)
    T9 = fd.ops.add(T5, T8)
    S10 = fd.ops.size(T5, dim=0)
    S11 = fd.define_scalar(1, dtype=DataType.Int)
    V12 = fd.define_vector([S10, S11], dtype=DataType.Int)
    T13 = fd.ops.broadcast_in_dim(T1, shape=V12, broadcast_dims=[0])
    V14 = fd.ops.shape(T5)
    T15 = fd.ops.broadcast_in_dim(T13, shape=V14, broadcast_dims=[0, 1])
    T16 = fd.ops.sub(T9, T15)
    T17 = fd.ops.broadcast_in_dim(T2, shape=V14, broadcast_dims=[0, 1])
    T18 = fd.ops.mul(T16, T17)
    T19 = fd.ops.broadcast_in_dim(T4, shape=V14, broadcast_dims=[1])
    T20 = fd.ops.sum(T3, axes=[0], keepdim=False, dtype=DataType.Null)
    T21 = fd.ops.mul(T3, T19)
    T22 = fd.ops.mul(T3, T18)
    T23 = fd.ops.sum(T22, axes=[0], keepdim=False, dtype=DataType.Null)
    T24 = fd.ops.mul(T21, T17)
    T25 = fd.ops.mul(T21, T16)
    T26 = fd.ops.sum(T25, axes=[1], keepdim=False, dtype=DataType.Null)
    T27 = fd.ops.broadcast_in_dim(T26, shape=V12, broadcast_dims=[0])
    T28 = fd.ops.neg(T24)
    T29 = fd.ops.sum(T28, axes=[1], keepdim=False, dtype=DataType.Null)
    T30 = fd.ops.broadcast_in_dim(T29, shape=V12, broadcast_dims=[0])
    S31 = fd.define_scalar(-0.500000, dtype=DataType.Double)
    T32 = fd.ops.mul(S31, T27)
    S33 = fd.define_scalar(3.00000, dtype=DataType.Double)
    T34 = fd.ops.pow(T2, S33)
    T35 = fd.ops.mul(T32, T34)
    T36 = fd.ops.sum(T30, axes=[1], keepdim=False, dtype=DataType.Null)
    T37 = fd.ops.sum(T35, axes=[1], keepdim=False, dtype=DataType.Null)
    T38 = fd.ops.broadcast_in_dim(T37, shape=V12, broadcast_dims=[0])
    T39 = fd.ops.broadcast_in_dim(T38, shape=V14, broadcast_dims=[0, 1])
    T40 = fd.ops.broadcast_in_dim(T1, shape=V12, broadcast_dims=[0])
    T41 = fd.ops.broadcast_in_dim(T40, shape=V14, broadcast_dims=[0, 1])
    S42 = fd.define_scalar(2.00000, dtype=DataType.Double)
    T43 = fd.ops.mul(S42, T39)
    T44 = fd.ops.sub(T9, T41)
    T45 = fd.ops.mul(T43, T44)
    S46 = fd.ops.size(T5, dim=1)
    S47 = fd.ops.reciprocal(S46)
    T48 = fd.ops.mul(T45, S47)
    T49 = fd.ops.broadcast_in_dim(T36, shape=V12, broadcast_dims=[0])
    T50 = fd.ops.broadcast_in_dim(T49, shape=V14, broadcast_dims=[0, 1])
    T51 = fd.ops.mul(S47, T50)
    T52 = fd.ops.add(T48, T51)
    T53 = fd.ops.add(T24, T52)
    T54 = fd.ops.mul(T53, S7)
    T55 = fd.ops.mul(T54, T0)
    T56 = fd.ops.add(T53, T55)
    fd.add_output(T56)
    fd.add_output(T23)
    fd.add_output(T20)

with FusionDefinition() as fd:
    nvfuser_fusion_id0(fd)

inputs = [
    torch.randint(0, 2, (659456,), dtype=torch.bool, device='cuda:0').as_strided((16, 41216), (41216, 1)),
    torch.randn((16,), dtype=torch.float32, device='cuda:0').as_strided((16,), (1,)),
    torch.randn((16,), dtype=torch.float32, device='cuda:0').as_strided((16, 1), (1, 1)),
    torch.randn((659456,), dtype=torch.float32, device='cuda:0').as_strided((16, 41216), (41216, 1)),
    torch.randn((41216,), dtype=torch.float32, device='cuda:0').as_strided((41216,), (1,)),
    torch.randn((659456,), dtype=torch.float32, device='cuda:0').as_strided((16, 41216), (41216, 1)),
]
fd.execute(inputs)
```

```
Traceback (most recent call last):
  File "/opt/pytorch/nvfuser/nvfuser/__init__.py", line 137, in execute
    result = self._execute(
RuntimeError: it != id_to_kernel_runtime_.end() INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/csrc/kernel_cache.cpp":630, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. 
Exception raised from evictCache at /opt/pytorch/nvfuser/csrc/kernel_cache.cpp:630 (most recent call first):
frame #0: nvfuser::nvfCheckFail(char const*, char const*, unsigned int, char const*) + 0xe8 (0x7fed177d6ba5 in /opt/pytorch/nvfuser/nvfuser/lib/libnvfuser_codegen.so)
frame #1: nvfuser::FusionExecutorCache::evictCache(unsigned long) + 0x81 (0x7fed17a86381 in /opt/pytorch/nvfuser/nvfuser/lib/libnvfuser_codegen.so)
frame #2: nvfuser::FusionExecutorCache::prepareInputs(c10::ArrayRef<c10::IValue> const&, std::optional<signed char>) + 0xa7 (0x7fed17a8bfa7 in /opt/pytorch/nvfuser/nvfuser/lib/libnvfuser_codegen.so)
frame #3: nvfuser::FusionExecutorCache::runFusionWithInputs(c10::ArrayRef<c10::IValue> const&, std::optional<nvfuser::PrimDataType>, std::optional<signed char>) + 0x1da (0x7fed17a984ca in /opt/pytorch/nvfuser/nvfuser/lib/libnvfuser_codegen.so)
frame #4: nvfuser::python_frontend::FusionDefinition::execute(c10::ArrayRef<c10::IValue> const&, bool, bool, std::optional<signed char>) const + 0x16c (0x7fed17da675c in /opt/pytorch/nvfuser/nvfuser/lib/libnvfuser_codegen.so)
```

**Comment by jacobhinkle** [comment]
I can't seem to reproduce this on my machine. Does this fail when you use the repro that is printed in the error message or should I run the whole benchmark to reproduce?

**Comment by Priya2698** [comment]
> I can't seem to reproduce this on my machine. Does this fail when you use the repro that is printed in the error message or should I run the whole benchmark to reproduce?

This one is occurring when running the whole benchmark. Updated the issue to note this.

**Comment by jacobhinkle** [comment]
Thanks. As a side note, on my 24GB 3090 Ti I get an OOM when running `python_benchmarks/test_batchnorm_bwd.py::test_batchnorm_bwd_benchmark[True-torch.float32-(256, 256, 112, 112)]`. This is not a cache-clearing issue since it happens when I run this test directly. This might be fine if we only run benchmarks on machines with 40GB available.

**Comment by Priya2698** [comment]
The benchmarks are run on 80GB machines so the sizes are set according to that.

**Comment by naoyam** [comment]
> The benchmarks are run on 80GB machines so the sizes are set according to that.

Can you please make sure the benchmarks won't fail on devices with a smaller memory capacity? Can we adjust the size based on the available memory of the actual device?

**Comment by Priya2698** [comment]
> > The benchmarks are run on 80GB machines so the sizes are set according to that.
> 
> Can you please make sure the benchmarks won't fail on devices with a smaller memory capacity? Can we adjust the size based on the available memory of the actual device?

One way could be to wrap the validation and the benchmark run in a `try-except` block to skip the test when we exceed the memory of the device.
I'll look into variably setting the input sizes based on device capacity: since different benchmarks have different memory requirements we will have to do this at an individual benchmark level.

**Comment by Priya2698** [comment]
This error seems to be resolved after #1607. 
I'll reopen this if the error is seen again.
